File: Algorithms/src/2125. Number of Laser Beams in a Bank/Code/solution.py
from typing import List

class Solution:
    def numberOfBeams(self, bank: List[str]) -> int:
        ans = 0         # total beams
        prev = 0        # device count in previous non-empty row
        
        for row in bank:
            cnt = row.count('1')   # count devices in current row
            if cnt > 0:
                ans += prev * cnt  # beams between previous non-empty row and current row
                prev = cnt         # update previous
        return ans

File: Algorithms/src/2125. Number of Laser Beams in a Bank/Code/solution.cpp
class Solution
{
public:
    int numberOfBeams(vector<string> &bank)
    {
        long long ans = 0;  // total beams (use long long to be safe)
        long long prev = 0; // device count in the previous non-empty row

        for (auto &row : bank)
        {
            long long cnt = 0;
            // count '1's in current row
            for (char ch : row)
                if (ch == '1')
                    ++cnt;

            if (cnt > 0)
            {
                // beams between this row and previous non-empty row
                ans += prev * cnt;
                prev = cnt; // update previous to current
            }
            // if cnt == 0, skip but prev remains unchanged
        }
        return (int)ans;
    }
};
